### DISCUSSION TITLE
How to define the executeable_path?

### DISCUSSION THREAD

**Discussion by QThans** [apify/crawlee-python#715] [Q&A] [answered]
Due to some reason, I hope the project can use a specific version:https://googlechromelabs.github.io/chrome-for-testing/

**Comment by Mantisus** [comment, marked as answer]
Hi!

You can use [`PlaywrightBrowserPlugin`](https://crawlee.dev/python/api/class/PlaywrightBrowserPlugin) for this purpose

Example:

```python
from crawlee.playwright_crawler import PlaywrightCrawler, PlaywrightCrawlingContext
from crawlee.browsers import BrowserPool, PlaywrightBrowserPlugin


user_plugin = PlaywrightBrowserPlugin(browser_options={"executable_path": "{PATH}"})

browser_pool = BrowserPool(plugins=[user_plugin])

crawler = PlaywrightCrawler(browser_pool=browser_pool)
```

**Reply by QThans** [reply]
thanks

**Reply by QThans** [reply]
> Hi!
> 
> You can use [`PlaywrightBrowserPlugin`](https://crawlee.dev/python/api/class/PlaywrightBrowserPlugin) for this purpose
> 
> Example:
> 
> ```python
> from crawlee.playwright_crawler import PlaywrightCrawler, PlaywrightCrawlingContext
> from crawlee.browsers import BrowserPool, PlaywrightBrowserPlugin
> 
> 
> user_plugin = PlaywrightBrowserPlugin(browser_options={"executable_path": "{PATH}"})
> 
> browser_pool = BrowserPool(plugins=[user_plugin])
> 
> crawler = PlaywrightCrawler(browser_pool=browser_pool)
> ```

It's right.
And, do u know how to set launchContext?

```
const crawler = new PuppeteerCrawler({
        proxyConfiguration,
        requestHandler: router,
        // maxRequestsPerCrawl: 3,
        minConcurrency: 3,
        maxConcurrency: 5,
        headless: headless,
        launchContext: {
            // Native Puppeteer options
            launchOptions: {
                args: ['--disable-features=TrackingProtection3pcd', '--disable-web-security', '--no-sandbox'],
            }
        },
    });
```
I hope can do as same as above code in crawlee-python.

**Reply by QThans** [reply]
> > Hi!
> > You can use [`PlaywrightBrowserPlugin`](https://crawlee.dev/python/api/class/PlaywrightBrowserPlugin) for this purpose
> > Example:
> > ```python
> > from crawlee.playwright_crawler import PlaywrightCrawler, PlaywrightCrawlingContext
> > from crawlee.browsers import BrowserPool, PlaywrightBrowserPlugin
> > 
> > 
> > user_plugin = PlaywrightBrowserPlugin(browser_options={"executable_path": "{PATH}"})
> > 
> > browser_pool = BrowserPool(plugins=[user_plugin])
> > 
> > crawler = PlaywrightCrawler(browser_pool=browser_pool)
> > ```
> 
> It's right. And, do u know how to set launchContext?
> 
> ```
> const crawler = new PuppeteerCrawler({
>         proxyConfiguration,
>         requestHandler: router,
>         // maxRequestsPerCrawl: 3,
>         minConcurrency: 3,
>         maxConcurrency: 5,
>         headless: headless,
>         launchContext: {
>             // Native Puppeteer options
>             launchOptions: {
>                 args: ['--disable-features=TrackingProtection3pcd', '--disable-web-security', '--no-sandbox'],
>             }
>         },
>     });
> ```
> 
> I hope can do as same as above code in crawlee-python.

user_plugin = PlaywrightBrowserPlugin(browser_options={
    "headless":False,
    "args": ["--no-sandbox","--disable-features=TrackingProtection3pcd","--disable-web-security"],
    "executable_path": '{PATH}',
})

**Comment by vdusek** [comment]
Hi, you can provide the `executable_path` as well as other browser launch options via the `browser_options` argument to the specific plugin (Playwright). The `browser_options` is currently not exposed on the Crawler level in the Python implementation (only the `headless` and `browser_type`), but we will soon do it (#719).

I will add that setting these values via env variables is currently not supported. You can monitor the progress in #670.
